### PR TITLE
REPO-5142: Refactored events2 code to allow extension.

### PR DESCRIPTION
**Pull request by jamalkm**
- Refactored events2 code to allow extension.
- Fixed issues caused by the acs-event-model refactoring work.

### CODE REVIEW

File: src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java [hunk 1]
```diff
@@ -37,10 +37,12 @@
 import org.alfresco.model.ContentModel;
 import org.alfresco.repo.event.databind.ObjectMapperFactory;
```
Comment (SaraAspery) [inline src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java:38]: unused import

File: src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java [hunk 2]
```diff
@@ -37,10 +37,12 @@
...
 import org.alfresco.repo.event.v1.model.ChildAssociationResource;
+import org.alfresco.repo.event.v1.model.DataAttributes;
 import org.alfresco.repo.event.v1.model.EventData;
```
Comment (SaraAspery) [inline src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java:41]: unused import

File: src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java [hunk 3]
```diff
@@ -497,7 +498,6 @@ public void testUpdateNodeTypeWithCustomType()
 
         NodeResource resourceBefore = getNodeResourceBefore(3);
         assertEquals("Incorrect node type was found", "cm:content", resourceBefore.getNodeType());
-        // assertNotNull(resourceBefore.getModifiedAt()); uncomment this when the issue will be fixed
```
Comment (SaraAspery) [inline src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java:500]: maybe add a TODO so it doesn't get missed?

File: src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java (hunk 2 above)
Comment (jamalkm) [reply src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java]: Done!

File: src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java (hunk 1 above)
Comment (jamalkm) [reply src/test/java/org/alfresco/repo/event2/AbstractContextAwareRepoEvent.java]: Done!

File: src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java (hunk 3 above)
Comment (jamalkm) [reply src/test/java/org/alfresco/repo/event2/UpdateRepoEventIT.java:500]: If you remember we closed the raised issue as "Not a Bug", so there is no need for the commented code or TODO.

**Pull request merged**